commit 71db5eb99c960e9c30e4b3ed04103c513b6251b5
Author: Tejun Heo <tj@kernel.org>
Date:   Wed Mar 23 10:37:00 2011 +0100

    signal: Remove superflous try_to_freeze() loop in do_signal_stop()
    
    do_signal_stop() is used only by get_signal_to_deliver() and after a
    successful signal stop, it always calls try_to_freeze(), so the
    try_to_freeze() loop around schedule() in do_signal_stop() is
    superflous and confusing.  Remove it.
    
    Signed-off-by: Tejun Heo <tj@kernel.org>
    Acked-by: Rafael J. Wysocki <rjw@sisk.pl>
    Acked-by: Oleg Nesterov <oleg@redhat.com>
    Acked-by: Roland McGrath <roland@redhat.com>
    [dannf: backported to Debian's 2.6.32]

diff --git a/kernel/signal.c b/kernel/signal.c
index 4a238e1..adc676d 100644
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -1772,9 +1772,7 @@ static int do_signal_stop(int signr)
 	}
 
 	/* Now we don't run again until woken by SIGCONT or SIGKILL */
-	do {
-		schedule();
-	} while (try_to_freeze());
+	schedule();
 
 	tracehook_finish_jctl();
 	current->exit_code = 0;
